### PR TITLE
Replace kotlin Pair with `DataFeed` data class

### PR DESCRIPTION
**Pull request by patrickkuo**

### CODE REVIEW

**Comment by mikehearn** [comment]
Do we really need to rename the RPCs too? Seems unnecessary and without that much code (that relies on type inference) will continue working with just a recompile.

**Comment by patrickkuo** [comment]
Don't really need to but the api looks a bit inconsistent, we have `stateMachinesAndUpdates`, `verifiedTransactions` and `networkMapUpdates` all returning similar thing (snapshot+updates). I didn't remove the renamed method so much of the code should still work(with deprecated warning). It will be nice to make them consistent before v1?

**Comment by mikehearn** [comment]
Well, fair enough I guess. You're right, the prior naming was inconsistent.

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 1]
```diff
@@ -382,3 +392,14 @@ inline fun <T : Any, A, B, C, D, reified R : FlowLogic<T>> CordaRPCOps.startTrac
         arg2: C,
         arg3: D
 ): FlowProgressHandle<T> = startTrackedFlowDynamic(R::class.java, arg0, arg1, arg2, arg3)
+
+/**
+ * The Data feed contains a snapshot of the requested data and a [Observable] for future update.
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: "an [Observable] of future updates"

File: core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt [hunk 2]
```diff
@@ -147,31 +147,41 @@ interface CordaRPCOps : RPCOps {
...
-    fun vaultAndUpdates(): Pair<List<StateAndRef<ContractState>>, Observable<Vault.Update>>
+    fun vaultAndUpdates(): DataFeed<List<StateAndRef<ContractState>>, Vault.Update>
 
     /**
-     * Returns a pair of all recorded transactions and an observable of future recorded ones.
+     * Returns a data feed of all recorded transactions and an observable of future recorded ones.
      */
     @RPCReturnsObservables
-    fun verifiedTransactions(): Pair<List<SignedTransaction>, Observable<SignedTransaction>>
+    fun verifiedTransactionFeed(): DataFeed<List<SignedTransaction>, SignedTransaction>
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/messaging/CordaRPCOps.kt]: verifiedTransactionsFeed

**Pull request merged**